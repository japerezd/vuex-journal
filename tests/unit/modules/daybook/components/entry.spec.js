import { shallowMount } from '@vue/test-utils';
import Entry from '@/modules/daybook/components/Entry.vue';
import daybookRouter from '@/modules/daybook/router';

describe('Pruebas en Entry component', () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(Entry, {
    props: {
      entry: {
        id: '-MiJYhNIHihlBvVuDr_5',
        date: 1630284981723,
        text: 'segunda entrada',
      },
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });
  test('debe de hacer match con el snapshot ', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('debe de redireccionar al hacer click en el entry-container', () => {
      const container = wrapper.find('.entry-container')
      container.trigger('click');
        expect(mockRouter.push).toHaveBeenCalledWith({
            name: 'entry',
            params: {
                id: wrapper.props().entry.id
            }
        });
  });

  test('pruebas en las propiedades computadas', () => {
    //   console.log(wrapper.vm)
    expect(wrapper.vm.day).toBe(29);
    expect(wrapper.vm.month).toBe('Agosto');
    expect(wrapper.vm.yearDay).toBe('2021, Domingo');
  });
});
