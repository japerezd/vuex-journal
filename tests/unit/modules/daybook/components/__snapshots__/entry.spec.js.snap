// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry component debe de hacer match con el snapshot  1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">29</span><span class="mx-1 fs-5">Agosto</span><span class="mx-2 fw-light">2021, Domingo</span></div>
  <div class="entry-description">segunda entrada</div>
</div>
`;
